src/lib.rs

```diff
@@ -8,34 +8,57 @@ use core::cmp::Ordering;
 
 use proc_macro2::{Ident, TokenStream};
 use quote::{format_ident, quote, ToTokens};
-use syn::parse::{Parse, ParseStream};
-use syn::punctuated::Punctuated;
-use syn::spanned::Spanned;
 use syn::{
-    Data, DeriveInput, Error, Fields, FieldsNamed, FieldsUnnamed, Result, Token, TraitBound, Type,
+    parse::{discouraged::Speculative, Parse, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    Data, DeriveInput, Error, Fields, FieldsNamed, FieldsUnnamed, Result, Token, Type,
 };
 
 /// Holds parsed [bounds](TraitBound) and [traits](Traits).
-struct DeriveWhere {
-    /// Parsed [bounds](TraitBound).
-    bounds: Vec<TraitBound>,
-    /// Parsed [traits](Traits).
-    traits: Vec<Traits>,
+enum DeriveWhere {
+    /// Generic type parameters were defined.
+    WithBounds(Vec<Type>, Vec<Traits>),
```

Comment:
I did that already in #15, we now that we have switched to the `Type` type, we shouldn't use the name `bounds` anymore, as it is wrong. Generic type parameters is the correct term, I propose using the variable name `generics`.